### PR TITLE
horizon/docs: Improve dev docs

### LINKED ISSUE #835: Improve Horizon dev docs
It would be cool if the docs provided two distinct and clear paths for people looking to
1) run Horizon/core in production
2) get in and start contributing on the dev side

We should look through the existing docs and see what can be combined, simplified or enhanced to make the doc design more user-oriented, and the barrier to entry for these use cases lower.

### PR DESCRIPTION
**Pull request by ire-and-curses**
This PR closes #835 by adding two new docs to make it easier for new Horizon users to get moving, and tidying up and de-duplicating some of the existing docs. This PR does not fix all the documentation issues, but it attacks some low-hanging fruit. It adds three clear linear flows for three different use cases.

* [README.md](https://github.com/stellar/go/blob/5db43729ff9cc9c7e58ca10b684f7645c90862f1/services/horizon/README.md) - The Horizon README is the root doc, linking out to three linear paths for interacting with the codebase. Build info is removed here.
*[developing.md](https://github.com/stellar/go/blob/2672b1ca92c2cc7f73574db14dfcd82c2c20960b/services/horizon/internal/docs/developing.md) - The new developer guide describes in detail how to get to a working local install, with tests that run and a local server that can be inspected. It's the doc I wish I'd had when I started working with Horizon.
*[quickstart.md](https://github.com/stellar/go/blob/eb74baa72dcf9ab74a96b8ecfb5a5c4243f193d8/services/horizon/internal/docs/quickstart.md) - A deliberately minimal doc aimed at getting the fastest possible path to a working Horizon + Stellar core for testing.
*[admin.md](https://github.com/stellar/go/blob/2672b1ca92c2cc7f73574db14dfcd82c2c20960b/services/horizon/internal/docs/admin.md) - The admin guide needs substantial work that is not addressed here. Changes were minimal, just linking to the other docs.
*[notes_for_developers.md](https://github.com/stellar/go/blob/eb74baa72dcf9ab74a96b8ecfb5a5c4243f193d8/services/horizon/internal/docs/notes_for_developers.md) - The existing developing guide has been renamed and moved. It needs expanding and cleaning up, but that is out of scope for this PR.

I've learned a lot about the docs during this process and have a few thoughts about next steps. I'd be happy to discuss a broader plan for improving them going forward, if that becomes a priority.

### CODE REVIEW

File: services/horizon/internal/docs/developing.md [hunk 1]
```diff
@@ -1,181 +1,96 @@
...
 
-start a redis server on port `6379`
+1. Install [Docker](https://www.docker.com/get-started).
+2. Verify your Docker installation works: `docker run hello-world`
+3. Create a local directory that the container can use to record state. This is helpful because it can take a few minutes to sync a new `stellar-core` with enough data for testing, and because it allows you to inspect and modify the configuration if needed. Here, we create a directory called `stellar` to use as the persistent volume: `cd $HOME; mkdir stellar`
+4. Download and run the Stellar Quickstart container, replacing `USER` with your username:
 
 ```bash
-redis-server
+docker run --rm -it -p "8000:8000" -p "11626:11626" -p "11625:11625" -p"8002:5432" -v /home/USER/stellar:/opt/stellar --name stellar stellar/quickstart --testnet
```
Comment (tomquisel) [inline services/horizon/internal/docs/developing.md]: Maybe just replace `/home/USER` with `$HOME`? `$HOME` works on mac while the other doesn't

File: services/horizon/internal/docs/developing.md [hunk 2]
```diff
@@ -1,181 +1,96 @@
...
 ```
 
-or run individual Horizon tests like so, providing the expected arguments:
-
+## Check Stellar Core status
+Stellar Core takes some time to synchronise with the rest of the network. The default configuration will pull roughly a couple of day's worth of ledgers, and may take 15 - 30 minutes to catch up. You can check the progress by monitoring logs:
 ```bash
-go test github.com/stellar/go/services/horizon/...
+docker exec -it stellar /bin/bash
+tail -f var/log/supervisor/stellar-core-stdout---supervisor-aywkVK.log
```
Comment (tomquisel) [inline services/horizon/internal/docs/developing.md]: Does this `tail` command work even with the random string at the end of the log file? Maybe `...supervisor-*.log` would work more generally?

File: services/horizon/internal/docs/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+The fastest way to get up and running is using the [Stellar Quickstart Docker Image](https://github.com/stellar/docker-stellar-core-horizon). This is a Docker container that provides both `stellar-core` and `horizon`, pre-configured for testing.
+
+1. Install [Docker](https://www.docker.com/get-started).
+2. Verify your Docker installation works: `docker run hello-world`
+3. Create a local directory that the container can use to record state. This is helpful because it can take a few minutes to sync a new `stellar-core` with enough data for testing, and because it allows you to inspect and modify the configuration if needed. Here, we create a directory called `stellar` to use as the persistent volume:
+`cd $HOME; mkdir stellar`
+4. Download and run the Stellar Quickstart container, replacing `USER` with your username:
+
+```bash
+docker run --rm -it -p "8000:8000" -p "11626:11626" -p "11625:11625" -p"8002:5432" -v /home/USER/stellar:/opt/stellar --name stellar stellar/quickstart --testnet
```
Comment (tomquisel) [inline services/horizon/internal/docs/quickstart.md]: `/home/USER` -> `$HOME`

File: services/horizon/internal/docs/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+You can check out Stellar Core status by browsing to http://localhost:11626.
+
+You can check out your Horizon instance by browsing to http://localhost:8000.
+
+You can tail logs within the container to see what's going on behind the scenes:
+```bash
+docker exec -it stellar /bin/bash
+tail -f var/log/supervisor/stellar-core-stdout---supervisor-aywkVK.log
```
Comment (tomquisel) [inline services/horizon/internal/docs/quickstart.md]: The log filename comment again (`aywkVK`)

File: services/horizon/internal/docs/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+tail -f var/log/supervisor/horizon-stderr---supervisor-lkZnZ5.log
+```
+
+On a modern laptop this test setup takes about 15 minutes to synchronise with the last couple of days of testnet ledgers. At that point Horizon will be available for querying.
```
Comment (tomquisel) [inline services/horizon/internal/docs/quickstart.md]: Is there a way to configure the docker to load less history and so start up faster?

File: services/horizon/internal/docs/admin.md [hunk 6]
```diff
@@ -1,12 +1,12 @@
----
-title: Administration
----
+## Horizon Administration Guide
```
Comment (bartekn) [inline services/horizon/internal/docs/admin.md]: Metadata section in the top (`title`) is used by [metalsmith](https://metalsmith.io/) that generates [the dev portal](https://github.com/stellar/developers). Let's leave them.

File: services/horizon/internal/docs/developing.md (hunk 2 above)
Comment (bartekn) [reply services/horizon/internal/docs/developing.md]: Users can also use [`supervisorctl`](http://supervisord.org/running.html#running-supervisorctl) to print logs. It works without random strings, command like `tail -f horizon` would continuously print the Horizon logs.

File: services/horizon/internal/docs/developing.md [hunk 7]
```diff
@@ -1,181 +1,96 @@
...
-log.WithField("init_name", i.Name).Info("running initializer")
+```bash
+horizon --db-url="postgres://localhost/horizon_dev" --stellar-core-db-url="postgres://stellar:postgres@localhost:8002/core" --stellar-core-url="http://localhost:11626" --port 8001 --network-passphrase "Test SDF Network ; September 2015" --ingest
 ```
 
-With the "bad" form of the logging example above, an operator can filter on both the message as well as the initializer name independently.  This gets more powerful when multiple fields are combined, allowing for all sorts of slicing and dicing.
-
-
-## <a name="TLS"></a> Enabling TLS on your local workstation
+If all is well, you should see ingest logs written to standard out. You can test your Horizon instance with a query like: http://localhost:8001/transactions?cursor=&limit=10&order=asc. Use the [Stellar Laboratory](https://www.stellar.org/laboratory/) to craft other queries to try out,
```
Comment (bartekn) [inline services/horizon/internal/docs/developing.md]: We can omit empty params (`cursor=`).

File: services/horizon/internal/docs/quickstart.md (hunk 5 above)
Comment (bartekn) [reply services/horizon/internal/docs/quickstart.md]: Currently no but it should be easy to be able to pass the value via env variable or CLI param.

Comment (ire-and-curses) [reply services/horizon/internal/docs/quickstart.md]: I think you can? This is in persistent mode, so you can start the container, then stop it, and then edit `$HOME/stellar/core/etc/stellar-core.cfg` and change `CATCHUP_RECENT` to a number less than 8640. Do you want me to add something like this to the doc?

File: services/horizon/internal/docs/developing.md (hunk 2 above)
Comment (ire-and-curses) [reply services/horizon/internal/docs/developing.md]: `supervisorctl` is a nice solution!

File: services/horizon/internal/docs/quickstart.md (hunk 5 above)
Comment (bartekn) [reply services/horizon/internal/docs/quickstart.md]: Yeah, I was thinking about an easy way, like passing it via CLI param and internal script edits the config file for you.

File: services/horizon/internal/docs/admin.md (hunk 6 above)
Comment (ire-and-curses) [reply services/horizon/internal/docs/admin.md]: Ok. I wish they didn't look so bad when viewed from Github.

File: services/horizon/internal/docs/quickstart.md (hunk 5 above)
Comment (ire-and-curses) [reply services/horizon/internal/docs/quickstart.md]: Ok, I raised https://github.com/stellar/docker-stellar-core-horizon/issues/102 to track this. If someone with permissions wanted to add `help-wanted` tag that would be cool.

File: services/horizon/internal/docs/developing.md [hunk 8]
```diff
@@ -1,181 +1,99 @@
...
 
-1. [`stellar-core-commander`](https://github.com/stellar/stellar_core_commander) (in short: `scc`) installed and [configured](https://github.com/stellar/stellar_core_commander#assumptions-about-environment).
-2. [`stellar-core`](https://github.com/stellar/stellar-core) binary.
-3. This repository cloned locally.
+Open a new terminal. Confirm everything worked by running `horizon --help` successfully. You should see an informative message listing the command line options supported by Horizon.
 
-`scc` allows you to write scripts/recipes that are later executed in `stellar-core` isolated network. After executing a recipe you can then export `stellar-core` database to be able to run Horizon ingestion system against it (this repository contains a script that does this for you - read below).
+## Set up Horizon's database
+Horizon uses a Postgres database backend to store test fixtures and record information ingested from an associated Stellar Core. To set this up:
+1. Install [PostgreSQL](https://www.postgresql.org/).
```
[the remark this answers is not in the document]
Comment (ire-and-curses) [reply services/horizon/internal/docs/developing.md]: Created #853 to track this.

**Pull request merged**